jscomp/ml/ctype.ml

```diff
@@ -3701,6 +3701,14 @@ let rec subtype_rec env trace t1 t2 cstrs =
         with Exit ->
           (trace, t1, t2, !univar_pairs)::cstrs
         end
+    | (Tvariant {row_closed=true; row_fields}, Tconstr (_, [], _)) 
+        when extract_concrete_typedecl_opt env t2 |> Variant_coercion.type_is_variant -> 
```

Comment:
I wonder if there's a better way to do this. I had to check that it's an actual variant, or the case under would be blocked, which broke a coercion test.

Reply:
@cristianoc anything else we should check for here? Private?

Reply:
Private does not change the internal representation, so I would not bother.